sequencer/src/l1_client.rs

```diff
@@ -276,140 +162,54 @@ async fn get_finalized_block<P: JsonRpcClient>(
+        assert_eq!("anvil/v0.2.0", version);
 
-        let mut anvil = AnvilOptions::default()
-            .block_time(Duration::from_secs(1))
-            .spawn()
-            .await;
-        let l1_client = L1ClientOptions::with_url(anvil.ws_url())
-            .start()
-            .await
-            .unwrap();
+        // Test that nothing funky is happening to the provider when
+        // passed along in state.
+        let state = NodeState {
+            l1_client: L1Client::new(anvil.endpoint().parse().unwrap()),
+            ..Default::default()
+        };
+        let version = state.l1_client().provider.client_version().await.unwrap();
+        assert_eq!("anvil/v0.2.0", version);
 
-        // Spawn a task to check that updates to the snapshotted state are always increasing.
-        let (kill_safety_task, mut killed) = oneshot::channel();
-        let l1_client_clone = l1_client.clone();
-        let safety_task = spawn(async move {
-            let l1_client = l1_client_clone;
-            let mut prev_head = l1_client.snapshot().await.head;
-            loop {
-                if !matches!(killed.try_recv(), Ok(None)) {
-                    // The task was killed or the other end of the channel was dropped.
-                    return Ok(());
-                }
+        // compare response of underlying provider w/ `get_block_number`
+        let expected_head = provider.get_block_number().await.unwrap().as_u64();
+        let head = l1_client.get_block_number().await;
+        assert_eq!(expected_head, head);
 
-                // Ensure the snapshot has not decreased.
-                let head = l1_client.snapshot().await.head;
-                if head < prev_head {
-                    return Err(format!("head decreased from {prev_head} to {head}"));
-                }
-                prev_head = head;
-                sleep(Duration::from_millis(500)).await;
-            }
-        });
+        // compare response of underlying provider w/ `get_finalized_block`
+        let expected_finalized = provider.get_block(BlockNumber::Finalized).await.unwrap();
+        let finalized = l1_client.get_finalized_block().await.unwrap();
 
-        // Force an L1 reorg. Make it several blocks deep to be sure the L1 client task doesn't miss
-        // it while reconnecting.
-        anvil.reorg(5).await;
+        assert_eq!(expected_finalized.unwrap().hash.unwrap(), finalized.hash);
 
-        // Let the safety task run for a few more seconds to ensure we recover from the reorg and
-        // continue to make progress.
-        let snapshot = l1_client.snapshot().await.head;
-        sleep(Duration::from_secs(5)).await;
+        // If we drop `anvil` the same request will fail and we will
+        // get an error as below.
+        drop(anvil);
+        let error = provider.client_version().await.unwrap_err();
+        assert!(error
+            .to_string()
+            .contains("Connection reset by peer (os error 104)"));
```

Comment:
This error message varies across OSes, and is also subject to change in future versions of the anvil client. IMO what we're testing here is not important enough to justify such a brittle check. The `unwrap_err` above on its own should be sufficient

Reply:
fixed in 447115188afc4855b2a1f6e56535af1675dc9356